File: game.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.HashMap;
import java.util.List;
import java.util.Random;
import java.util.Scanner;
import java.util.concurrent.TimeUnit;
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;

public class Game {
    private int score;
    private List<String> words;
    private HashMap<String, Integer> leaderboard;
    private List<String> dictionary;

    public Game() throws IOException {
        this.score = 0;
        this.words = new ArrayList<>();
        this.leaderboard = new HashMap<>();
        this.dictionary = new ArrayList<>();
        BufferedReader reader = new BufferedReader(new FileReader("dictionary.txt"));
        String line = reader.readLine();
        while (line != null) {
            this.dictionary.add(line.trim());
            line = reader.readLine();
        }
        reader.close();
    }

    public void selectCategory() {
        System.out.println("What version would you like to play? :\n 1) General \n 2) Animals \n 3) Drinks \n 4) Foods \n 5) Music \n");
        // Prompt the user to select a category of words to play the game with
        while (true) {
            Scanner sc = new Scanner(System.in);
            String userNum = sc.nextLine();
            if (userNum.equals("1")) {
                this.words = this.dictionary;
                break;
            } else if (userNum.equals("2")) {
                this.words = readWordsFromFile("animals.txt");
                break;
            } else if (userNum.equals("3")) {
                this.words = readWordsFromFile("drinks.txt");
                break;
            } else if (userNum.equals("4")) {
                this.words = readWordsFromFile("foods.txt");
                break;
            } else if (userNum.equals("5")) {
                this.words = readWordsFromFile("music.txt");
                break;
            } else {
                System.out.println("That was not an option. Try again.");
            }
        }
    }

    private List<String> readWordsFromFile(String fileName) {
        List<String> words = new ArrayList<>();
        try {
            BufferedReader reader = new BufferedReader(new FileReader(fileName));
            String line = reader.readLine();
            while (line != null) {
                words.add(line.trim());
                line = reader.readLine();
            }
            reader.close();
        } catch (IOException e) {
            System.out.println("Error reading words from file: " + e);
        }
        return words;
    }

    public void play(String username) {
        // Set the initial score to 0
        score = 0;
        // Set the initial word length to 3
        int wordLength = 3;
        // Set the number of correct guesses required to increment word length to 5
        int correctGuessesNeeded = 5;
        // Initialize the correct guess counter
        int correctGuesses = 0;
        int incorrectGuesses = 0;
        int duration = 60;
        long startTime = System.currentTimeMillis();
        long endTime = startTime + TimeUnit.SECONDS.toMillis(duration);
        List<String> usedWords = new ArrayList<>();

        // Loop while there is still time left
        while (true) {
            // Select a random word of the current word length from the dictionary
            List<String> wordsOfCurrentLength = new ArrayList<>();
            for (String word : words) {
                if (word.length() == wordLength && !usedWords.contains(word)) {
                    wordsOfCurrentLength.add(word);
                }
            }
            Random rand = new Random();
            String word = wordsOfCurrentLength.get(rand.nextInt(wordsOfCurrentLength.size()));
            usedWords.add(word);

           

File: game.py
import time
class Game:
    def __init__(self):
        self.score = 0
        self.word_length = 3
        self.correct_guesses = 0
        self.correct_needed = 5
    def play(self):
        start_time = time.time()
        end_time = start_time + 60

        # Loop while there is still time left
        while True:
            # Select a random word of the current word length from the dictionary
            word = random.choice([w for w in chosen_list if len(w) == word_length])

            # Scramble the word
            scrambled_word = ''.join(random.sample(word, len(word)))

            while scrambled_word == word:
                scrambled_word = ''.join(random.sample(word, len(word)))

            # Display the scrambled word to the user
            print('Scrambled word:', scrambled_word)

            # Read the user's guess for the unscrambled word
            guess = input('Your guess: ')

            # If the guess is correct
            if guess == word:
                # Add the length of the unscrambled word times 10 to the score
                self.score += len(word) * 10

                # Add a fixed amount of time to the remaining time
                self.end_time += 3

                # Increment the correct guess counter
                self.correct_guesses += 1

                # If the correct guess counter reaches the required number
                if self.correct_guesses == self.correct_needed:
                    # Increment the word length
                    self.word_length += 1

                    # Reset the correct guess counter
                    correct_guesses = 0

            # If the guess is incorrect
            else:
                # Subtract the length of the scrambled word times 10 from the score
                self.score -= len(scrambled_word) * 10
                print(f"Wrong. One correct answer was {word}.")
                if self.score < 0:
                    self.score = 0

                # Subtract a fixed amount of time from the remaining time
                #time_left -= 5

            # Print the remaining time and current score
            current_time = time.time()
            time_remaining = max(0, int(end_time - current_time))
            print(f"Time remaining: {time_remaining} seconds")

            if time_remaining == 0:
                print("Countdown complete!")
                break

            time.sleep(1)  # Sleep for 1 second
            print('Current score:', self.score)

        # Display the final score to the user
        print('Game over! Your final score is:', self.score)
